tsparser/src/builder/package_mgmt.rs

```diff
@@ -16,14 +16,38 @@ struct PackageJson {
     dependencies: HashMap<String, String>,
 }
 
+fn parse_package_json(package_json_path: &Path) -> Result<PackageJson> {
+    let package_json = std::fs::read_to_string(package_json_path)
+        .with_context(|| format!("failed to read {}", package_json_path.display()))?;
+
+    serde_json::from_str(&package_json)
+        .with_context(|| format!("failed to parse {}", package_json_path.display()))
+}
+
+fn find_workspace_package_manager(mut dir: PathBuf) -> Result<Option<String>> {
+    while dir.pop() {
+        let package_json_path = dir.join("package.json");
+        if package_json_path.exists() {
+            let package_json = parse_package_json(&package_json_path)?;
+            if package_json.package_manager.is_some() {
```

Comment:
Should we check that it's not the empty string as well?

Reply:
update to continue the searching if the package manager field is empty :heavy_check_mark: